zkml/src/layers/mod.rs

```diff
@@ -307,7 +324,10 @@ impl Evaluate<Element> for Layer<Element> {
         inputs: &[&Tensor<Element>],
         unpadded_input_shapes: Vec<Vec<usize>>,
     ) -> Result<LayerOut<Element, E>> {
-        match self {
+        #[cfg(feature = "capture-layers-quant")]
+        let input_hash = sha256_hex(&(self, inputs));
```

Comment:
Maybe move this inside of `capture_quant::store`?

That would have a few benefits:

- Less code guarded by feature flags
- The sha256 can be computed over the data serialized there, so the hash matches the contents of the file
   - This also removes the need to serialize the data twice